### PR TITLE
Refactor structural reasoner

### PR DESCRIPTION
**Pull request by LckyLke**
Refactoring for Issue #98

### CODE REVIEW

**Comment by Demirrr** [comment]
Can you please add few tests ?

**Comment by alkidbaci** [comment]
I was hoping we could discuss this stuff about StructuralReasoner on Monday before doing any changes

**Comment by Demirrr** [comment]
Sorry, I don't think there is a need to discuss about these changes

File: owlapy/owl_reasoner.py [hunk 1]
```diff
@@ -80,36 +49,34 @@ def __init__(self, ontology: AbstractOWLOntology, *, class_cache: bool = True,
...
+        self.class_cache: bool = class_cache
+        self._property_cache: bool = property_cache
+        self._negation_default: bool = negation_default
+        self._sub_properties: bool = sub_properties
+        self.__warned: int = 0
         self._init()
 
-    def _init(self, cache_size=128):
-
-        individuals = self._ontology.individuals_in_signature()
```
Comment (alkidbaci) [inline owlapy/owl_reasoner.py:94]: I see that we are removing the line where we get the individuals once and we are adding individual retrieval on line 606, 647, 767, 803 which will be killing the algo in large KGs.

**Comment by alkidbaci** [comment]
> We haven't discussed using LRUCache but this seems to be integrated which I haven't seen it

I have not integrated LRUCache, it has been there since the time of Simon Bin and Lukas and I'm not against removing it

File: owlapy/owl_reasoner.py (hunk 1 above)
Comment (LckyLke) [reply owlapy/owl_reasoner.py:94]: Yes, i just replaced ind_set with the individuals_in_signature() method when ever it is used. 
Ig an advantage is that this is only computed if it is really need and not always on initialisation

Comment (Demirrr) [reply owlapy/owl_reasoner.py:94]: Let's get the correctness done first, then focus on trading of the memory with runtimes. 
It is my mistake that I should have not accept the PR based on caching.

**Pull request merged**